### PR TITLE
doc:netdev: mention get/set don't check for alignment of memory accessed

### LINKED ISSUE #4097: at86rf2xx: potential unaligned access
The NETDEV interface has lots of memory accesses like this:
```
*((uint16_t *)val) = at86rf2xx_get_txpower(dev);
```

I see that this is convenient, but can easily cause hardfaults because of unaligned access. Do we assume that the user only passes aligned memory if his platform doesn't support unaligned access? Either way this should be documented.

### PR DESCRIPTION
**Pull request by thomaseichinger**
fixes #4097

### CODE REVIEW

File: sys/include/net/gnrc/netdev.h [hunk 1]
```diff
@@ -118,6 +118,10 @@ typedef struct {
     /**
      * @brief   Get an option value from a given network device
      *
+     * @note    This function does not check for propor alignment of the memory
```
Comment (daniel-k) [inline sys/include/net/gnrc/netdev.h]: Typo: `propor` -> `proper`

File: sys/include/net/gnrc/netdev.h [hunk 2]
```diff
@@ -135,6 +139,10 @@ typedef struct {
     /**
      * @brief   Set an option value for a given network device
      *
+     * @note    This function does not check for propor alignment of the memory
```
Comment (daniel-k) [inline sys/include/net/gnrc/netdev.h]: Typo: `propor` -> `proper`

**Comment by thomaseichinger** [comment]
Addressed @daniel-k's comments and immediately squashed.

File: sys/include/net/gnrc/netdev.h [hunk 3]
```diff
@@ -1,4 +1,4 @@
-/*
+*
```
Comment (daniel-k) [inline sys/include/net/gnrc/netdev.h]: This doesn't look intended :smile:

Comment (thomaseichinger) [inline sys/include/net/gnrc/netdev.h]: Not at all, fixed.

**Comment by daniel-k** [comment]
Looks good, ACK from my side when squashed.

**Comment by thomaseichinger** [comment]
ACKed and Travis passed, go.

**Pull request merged**